galite-core/src/main/kotlin/org/kopi/galite/visual/ui/vaadin/form/DGridEditorColorField.kt

```diff
@@ -0,0 +1,121 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.kopi.galite.visual.ui.vaadin.form
+
+import java.awt.Color
+
+import org.kopi.galite.visual.form.VColorField
+import org.kopi.galite.visual.form.VFieldUI
+import org.kopi.galite.visual.ui.vaadin.base.BackgroundThreadHandler.access
+import org.kopi.galite.visual.ui.vaadin.grid.GridEditorColorField
+
+import com.vaadin.flow.data.binder.Result
+import com.vaadin.flow.data.binder.ValueContext
+import com.vaadin.flow.data.converter.Converter
+import com.vaadin.flow.data.renderer.Renderer
+
+class DGridEditorColorField(columnView: VFieldUI,
+                            label: DGridEditorLabel?,
+                            align: Int,
+                            options: Int) : DGridEditorField<Any?>(columnView, label, align, options) {
+
+  //---------------------------------------------------
+  // DATA MEMBERS
+  //---------------------------------------------------
+
+  private var color: String? = null
+
+  init {
+    (editor as GridEditorColorField).addValueChangeListener { this }
+    (editor as GridEditorColorField).getContent().element.addEventListener("change") {
+      setColortoVField((editor as GridEditorColorField).value as String?)
+    }
+  }
+
+  //---------------------------------------------------
+  // IMPLEMENTATION
+  //---------------------------------------------------
+  override fun updateText() {
+    setObject((getModel() as VColorField).getColor(columnView.blockView.getRecordFromDisplayLine(position)))
+  }
+
+  override fun getObject(): Any? = color
+
+  fun setObject(s: Color?) {
+    val stringColor = s?.let { colorToRgbString(s) }
+    access(currentUI) {
+      (editor as GridEditorColorField).setData(stringColor)
+    }
+    color = stringColor
+  }
+
+  override fun createEditor(): GridEditorColorField = GridEditorColorField()
+
+  override fun createConverter(): Converter<Any?, Any?> {
+    return object : Converter<Any?, Any?> {
+
+      override fun convertToModel(value: Any?, context: ValueContext?): Result<Any?>? = Result.ok(value)
+
+      override fun convertToPresentation(value: Any?, context: ValueContext?): Any? = value
+    }
+  }
+
+  override fun createRenderer(): Renderer<Any?> {
+    TODO()
+  }
+
+  override fun valueChanged(oldValue: String?) {
+    // Nothing to do
+  }
+
+  fun setColortoVField(value: String?) {
+    getModel().isChangedUI = true
+    getModel().setColor(rgbStringToColor(value))
+  }
+
+  /**
+   * Convert RGB String to Java.awt.Color.
+   */
+  private fun rgbStringToColor(hexString: String?): Color {
+    // Remove the '#' from the beginning of the RGB string
+    val rgbString = if (hexString!!.startsWith("#")) hexString.substring(1) else hexString
```

Comment:
We should consider the case of null values. If possible, change **hexString: String?** to **hexString: String**